### ISSUE TITLE
Support for sealed class hierarchies in ViewState

### ISSUE THREAD

**Issue by semanticer** [dimsuz/diff-dispatcher#3]
I was surprised when I discovered this isn't currently possible. Sealed class hierarchies are very often part of my ViewStates, especially if I work with some unidirectional architecture etc.

**Comment by dimsuz** [comment]
Can you post an example of how such `ViewState` would look? Do you use something [like this](https://github.com/iPoli/iPoli-android/blob/0c087aec97ae2ccf3f4dfbf22f2b53e4c6b84be4/app/src/main/java/mypoli/android/store/powerup/PowerUpListViewState.kt#L94)?

Also how does your typical `render` method usually looks like for `ViewState` modeled with sealed classes? Example would help me get a better idea on what to generate.

**Comment by semanticer** [comment]
Sure I have this situation: (Intent is not Android Intent)


```kotlin
interface State
@DiffElement(diffReceiver = IntentStateRenderer::class)
data class IntentViewState(val intents: List<Intent>): State

sealed class Intent
data class MultipleAppsIntent(val apps: List<App>): Intent()
data class SingleActionIntent(val action: String): Intent()

interface IntentStateRenderer {
    fun renderIntents(intents: List<Intent>)
}
```


In this case build fails, it complains there is no getIntents function in IntentViewState

**Comment by dimsuz** [comment]
Hmm. That sounds like a bug of some sort. Thanks, I'll investigate!

**Comment by dimsuz** [comment]
Fixed in `0.9.3` release.
